src/figuresClasses.ts

```diff
@@ -1,19 +1,78 @@
-export interface Figure {
+enum FigureShape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
+}
+
+type FigureColor = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  color: FigureColor,
+  shape: FigureShape,
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = FigureShape.triangle;
+
+  constructor(
+    public color: FigureColor,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Values cannot be <= 0');
+    }
 
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('The triangle does not exist');
+    }
+  }
+
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
```

Comment:
Here will be the number in any case. So you don't need to specify the type here. And better to use more meaningful naming. What does the `p` mean? 
```suggestion
    const p = (this.a + this.b + this.c) / 2;
```